file_util/inc/leatherman/file_util/file.hpp

```diff
@@ -12,6 +12,12 @@
 
 namespace leatherman { namespace file_util {
 
+    class file_error : public std::runtime_error {
```

Comment:
should contain the actual error code, look at `std::error_code`

include file name